python/tvm/hybrid/parser.py

```diff
@@ -0,0 +1,292 @@
+"""Compiling a subset of Python to HalideIR"""
+#pylint: disable=no-else-return
+import ast
+import operator
+import sys
+from ._internal import NOP, TRUE, RANGE_ONE, HALIDE_IMM, ZERO
+from ._intrin import LOOP_INTRIN, MATH_INTRIN
+from .var_decl import determine_variable_usage
+from ..api import thread_axis
+from .. import expr as _expr
+from .. import make as _make
+from .. import intrin
+from .. import api  as _api
+from .. import ir_pass as _ir_pass
+
+def list_to_block(visit, lst):
+    """Convert a list of Python IR nodes to HalideIR Block"""
+    lst = list(map(visit, lst))
+    lst = [stmt for stmt in lst if not _ir_pass.Equal(stmt, NOP)]
+    if not lst:
+        return NOP
+    if len(lst) == 1:
+        return lst[0]
+    body = lst[0]
+    for i in lst[1:]:
+        body = _make.Block(body, i)
+    return body
+
+class PyAST2HalideIR(ast.NodeVisitor):
+    """Python AST visitor pass which finally lowers it to HalideIR"""
+
+    _binop_maker = {
+        ast.Add   : operator.add,
+        ast.Sub   : operator.sub,
+        ast.Mult  : operator.mul,
+        ast.Div   : _make.Div,
+        ast.Mod   : operator.mod,
+        ast.BitOr : operator.or_,
+        ast.BitAnd: operator.and_,
+        ast.BitXor: operator.xor,
+        ast.Gt    : operator.gt,
+        ast.GtE   : operator.ge,
+        ast.Lt    : operator.lt,
+        ast.LtE   : operator.le,
+        ast.Eq    : operator.eq,
+        ast.NotEq : operator.ne,
+    }
+
+    _unaryop_maker = {
+        ast.USub   : operator.neg,
+        ast.Invert : operator.invert,
+        ast.Not    : operator.not_
+    }
+
+    def __init__(self, args, usage, func_name=None):
+        """
```

Comment:
documentation of init should go to class docstring, see https://github.com/dmlc/tvm/blob/master/python/tvm/contrib/rpc/tracker.py#L319